src/core/qgssnappingutils.cpp

```diff
@@ -387,7 +427,17 @@ void QgsSnappingUtils::defaultSettings( int& type, double& tolerance, QgsToleran
 
 const QgsCoordinateReferenceSystem* QgsSnappingUtils::destCRS()
 {
-  return mMapSettings.hasCrsTransformEnabled() ? &mMapSettings.destinationCrs() : 0;
+  if ( mSnapToType == SnapToType::SnapToMap )
+    return mMapSettings.hasCrsTransformEnabled() ? &mMapSettings.destinationCrs() : 0;
+  else if ( mSnapToType == SnapToType::SnapToLayer )
+    return &mCurrentLayer->crs();
```

Comment:
Hmm. This should probably also only be done if mMapSettings.hasCrsTransformEnabled()